src/operations/identify.rs

```diff
@@ -1,18 +1,33 @@
 use crate::{error::MagickError, image::Image};
 
+// https://imagemagick.org/script/command-line-options.php#identify
 pub fn identify(image: &mut Image) -> Result<(), MagickError> {
     println!("{}", identify_impl(image));
     Ok(())
 }
 
 fn identify_impl(image: &Image) -> String {
-    format!("{}x{}", image.pixels.width(), image.pixels.height())
+    let parts: Vec<String> = vec![
+        image.properties.filename.to_str().map(str::to_owned),
```

Comment:
This assumes that the filename can be converted to a &str, that is, that it can be represented in UTF-8. On many platforms this is not the case, and filenames can be any sequence of bytes. I've tried really hard to avoid that assumption and to use `OsString` for paths instead, and I'd like this code to handle that correctly too.

I've checked what `identify` does when encountering non-utf8 filenames, using [this guide](https://blog.augustfeng.app/articles/creating-files-in-linux-that-contains-non-utf-8-characters/) to create one, and it just repeats them verbatim.

So I believe the filename part will require special handling, without getting collected into a `Vec<String>`, so we can print out data that isn't valid UTF-8.

Reply:
Thanks for the hint, I had not really thought about that. I gave it a try, but I don't have a unix handy right now (I'm on macOS) so ... that's only what I could briefly make out using google on how that might possibly work. Basically I take all the non-filenames and convert them to `OsString`, which in this direction should be lossless, right? And then we print the plain bytes using the writer to stdout directly, never converting to `String`. I must admit though, https://doc.rust-lang.org/std/ffi/struct.OsString.html#method.as_encoded_bytes doesn't make it entirely clear to me that this is what we want. Maybe you had worked with that in the past?

Reply:
Oh, and everything _other_ than the filename is perfectly fine to collect into a `Vec<String>`. So you can just `write!` the filename and then work with the `Vec<String>` as usual.

Eventually we will have to add support for custom formatting templates where that's going to be trickier - but that's a problem for another day.

Reply:
> macOS doesn't allow non-utf8 filenames as far as I know and that's stated in the article as well.

Oh interesting, I missed that. Nevermind then, I'll add non-utf8 handling myself.

FYI, you can run a VM with Linux easily using https://mac.getutm.app/